boards: arm: add WeAct STM32F446 Core board

heronet opened the pull request:
# Add WeAct STM32F446 Core Board Support

## Overview
Adds support for the WeAct STM32F446 Core Board, a low-cost development board with STM32F446RET6 MCU.

## Hardware Features
- **MCU**: STM32F446RET6 (ARM Cortex-M4F @ 180MHz, 512KB Flash, 128KB SRAM)
- **Peripherals**: 4x UART, 4x I2C, 4x SPI, 2x CAN, 3x ADC, 2x DAC, USB OTG, SDIO
- **Onboard**: USB-C, User LED (PB2), User Button (PC13), MicroSD slot, SWD header
- **Clock**: 8MHz HSE, 32.768kHz LSE

## Testing
✅ **Samples tested:**
- `samples/basic/blinky` - LED control working
- `samples/hello_world` - UART console working  
- `samples/basic/button` - Button interrupts working

✅ **Flashing methods:**
- DFU via USB-C (hold BOOT0, press RESET)
- SWD debugging with ST-Link

## Board Availability
- Widely available on AliExpress/Taobao (~$10-15)
- [WeAct GitHub](https://github.com/WeActStudio/WeActStudio.STM32F4_64Pin_CoreBoard)
- Same pinout as other WeAct STM32F4 64-pin boards

**References:**
- [STM32F446 Datasheet](https://www.st.com/en/microcontrollers-microprocessors/stm32f446re.html)
- [Board Schematics](https://github.com/WeActStudio/WeActStudio.STM32F4_64Pin_CoreBoard/blob/master/Hardware/)

nordicjm commented on boards/weact/stm32f446_core/Kconfig.weact_stm32f446_core:
```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_WEACT_STM32F446_CORE
+    bool "WeAct Studio STM32F446 Core Board"
```
no prompt or type on here

nordicjm commented on boards/weact/stm32f446_core/Kconfig.weact_stm32f446_core:
```diff
@@ -0,0 +1,11 @@
+config BOARD_WEACT_STM32F446_CORE
+    bool "WeAct Studio STM32F446 Core Board"
+    select SOC_STM32F446XX
+    select SOC_SERIES_STM32F4X
+    select CPU_CORTEX_M4
+    select HAS_STM32CUBE
+    help
+      WeAct Studio STM32F446 Core Board based on STM32F446RE MCU.
```
```suggestion
```
these should be selected by the soc itself

nordicjm commented on boards/weact/stm32f446_core/doc/img/stm32f446_core.jpg:
trim whitespace, reduce resolution, convert to webp, put through https://tinypng.com/

nordicjm commented on boards/weact/stm32f446_core/weact_stm32f446_core.dts:
```diff
@@ -0,0 +1,132 @@
+
+	leds {
+		compatible = "gpio-leds";
+		led: led {
```
line gaps before nodes e.g. line 25, fix in whole pr

nordicjm commented on boards/weact/stm32f446_core/weact_stm32f446_core.dts:
```diff
@@ -0,0 +1,132 @@
+
+&rcc {
+  clocks = <&pll>;
+  clock-frequency = <DT_FREQ_M(180)>;
```
not using tab indent

nordicjm commented on boards/weact/stm32f446_core/weact_stm32f446_core_defconfig:
```diff
@@ -0,0 +1,16 @@
+
+CONFIG_SERIAL=y
+
+# console
```
capitalise first letter of comments

heronet reviewed (commented):
Update Kconfig and device tree for STM32F446 Core Board according to suggestions

heronet reviewed (commented):
update the image

heronet reviewed (commented):
change jpg image to webp

nordicjm commented:
commits need to be squashed

heronet reviewed (commented):
squashed commits

heronet reviewed (commented):
compliance check pass

heronet committed:
boards: arm: add WeAct STM32F446 Core board

Add support for WeAct STM32F446 Core board featuring:
- STM32F446RET6 MCU (ARM Cortex-M4F @ 180 MHz)
- 512KB Flash, 128KB SRAM
- 8MHz HSE crystal oscillator
- USB Type-C connector with DFU support
- User LED (PB2) and button (PC13)
- MicroSD card slot (SDIO)
- SWD debug header
- 30×2 pin headers exposing GPIO pins

Peripherals supported:
- 4x USART/UART
- 4x I2C
- 4x SPI/I2S
- 2x CAN
- 3x 12-bit ADC
- 2x 12-bit DAC
- USB OTG FS
- SDIO interface

Tested with samples/basic/blinky and samples/hello_world.

Signed-off-by: Siratul Islam <sirat4757@gmail.com>
Signed-off-by: heronet <sirat4757@gmail.com>

heronet commented:
Hi @kartben,
This PR has received 4 approvals and is ready to be merged. When you have a moment, could you please take a look and merge it if everything looks good?
Thank you for your time and help!

kartben commented:
> Hi @kartben, This PR has received 4 approvals and is ready to be merged. When you have a moment, could you please take a look and merge it if everything looks good? Thank you for your time and help!

it will be merged as soon as it's been in review for at least 2 business days (and if no one requests changes by then, of course). :-)
https://docs.zephyrproject.org/latest/project/project_roles.html#merge-criteria
